lib/ansible/module_utils/dimensiondata.py

```
@@ -0,0 +1,367 @@
+    user_id = None
+    key = None
+
+    # Attempt to grab from environment
+    if 'DIDATA_USER' in os.environ and 'DIDATA_PASSWORD' in os.environ:
+        user_id = os.environ['DIDATA_USER']
+        key = os.environ['DIDATA_PASSWORD']
+
+    # Environment failed try dot file
+    if user_id is None or key is None:
+        home = expanduser('~')
+        config = ConfigParser.RawConfigParser()
+        config.read("%s/.dimensiondata" % home)
+        try:
+            user_id = config.get("dimensiondatacloud", "DIDATA_USER")
+            key = config.get("dimensiondatacloud", "DIDATA_PASSWORD")
+        except:
+            pass
+
+    # Return False if either are not found
+    if user_id is None or key is None:
+        return False
+
+    # Both found, return data
+    return dict(user_id=user_id, key=key)
+
+
+# -------------------------
+# Get the list of available regions
+# whos vendor is Dimension Data.
+# -------------------------
+def get_dd_regions():
+    # Get endpoints
+    all_regions = API_ENDPOINTS.keys()
+
+    # filter to only DimensionData endpoints
+    region_list = filter(lambda i: i.startswith('dd-'), all_regions)
+
+    # Strip prefix
+    regions = [region[3:] for region in region_list]
+
+    return regions
+
+
+# ----------------------------------------
+# Get a network domain object by its name
+# ----------------------------------------
+def get_network_domain_by_name(driver, name, location):
+    networks = driver.ex_list_network_domains(location=location)
+    network = filter(lambda x: x.name == name, networks)
+    if len(network) > 0:
+        return network[0]
+    else:
+        return None
+
+
+# ---------------------------------------------
+# Get a network domain object by its name or id
+# ---------------------------------------------
+def get_network_domain(driver, locator, location):
+    if is_uuid(locator):
+        net_id = locator
+    else:
+        networks = driver.ex_list_network_domains(location=location)
```

don't all API calls need to capture exceptions?

---

Exception handling is done in module code. This way we can keep the handling of the error specific to the module using the shared code.

---

it seems inconsistent as you do capture it in some cases

---

Yes in some cases the error is always the same so we return False to ease module development. In other cases there are multiple error possibilities from API, some of which we ignore in one case and not another.

---

If the above is an issue we can make all return the exception and consistently handle them in the module.

---

+1 on raising a custom exception instead of failing silently.